client/src/components/Q&A/IndividualQuestion.jsx

```diff
@@ -2,15 +2,59 @@ import React, { useState } from 'react';
 import AnswersList from './AnswersList.jsx';
 import AddAnswerModal from './AddAnswerModal.jsx';
 import { formatDistanceToNow, parseISO } from 'date-fns';
+import axios from 'axios';
+import { validate } from 'react-email-validator';
 
 // Individual question - Integrate into QuestionsList.jsx
-const IndividualQuestion = ({ question, handleHelpful, handleReport }) => {
+const IndividualQuestion = ({ question, handleHelpful, handleReport, product }) => {
   // console.log('Individual question: ', question);
-
   const [isOpen, setIsOpen] = useState(false);
-
+  const [images, setImages] = useState([]);
   const { asker_name, question_body, question_helpfulness, question_date, question_id } = question;
 
+  const onClose = () => {
+    setIsOpen(false);
+  };
+
+  const photoWidget = cloudinary.createUploadWidget(
+    {
+      cloudName: 'dqk77sezi',
+      uploadPreset: 'FEC-add-photo'
+    },
+    (error, result) => {
+      if (!error && result && result.event === 'success') {
+        console.log('done! Here is the image info: ', result.info);
+        setImages([...images, result.info.url]);
+      }
+      if (error) { console.log(error); }
+    });
+
+  const handleSubmitAnswer = (e) => {
+    e.preventDefault();
+    const query = {
+      body: e.target.answer.value,
+      name: e.target.name.value,
+      email: e.target.email.value,
+      photos: images
+    };
+
+    if (!validate(query.email)) {
```

Comment:
I'm interested in how validate works here. Will it show the alert and also prevent the the post from happening?

Reply:
@joshgarza. OH speaking of validator, I forgot to add the "required" fields back into my form inputs.
This validator provides an alert, until the field is done correctly. Otherwise, it'll keep popping up. The POST won't run until the validator is resolved.

Reply:
That's really cool!